### ISSUE TITLE
configparser.SectionProxy.get* methods aren't correctly typed

### ISSUE THREAD

**Issue by OddBloke** [python/typeshed#1543]
On instantiation, `configparser.SectionProxy` takes as an argument the parser that it's a section of and uses `setattr` to populate itself with partially applied converters (i.e. methods that start with `get`) that the parser it is passed has.

Currently, this is stubbed as:

```py
class SectionProxy(MutableMapping[str, str]):
    def __init__(self, parser: RawConfigParser, name: str) -> None: ...

    <snip>

    # SectionProxy can have arbitrary attributes when custon converters are used
    def __getattr__(self, key: str) -> Callable[..., Any]: ...
```

which means that `ConfigParser()['section'].getboolean('key')` type checks as `Any`, whereas it should in fact be `bool`.

**Comment by OddBloke** [comment]
I think we can cover the majority of use cases by simply copying the definitions down from `RawConfigParser` and removing the first argument (`section`, which is partially applied at runtime).

This won't help people who are subclassing RawConfigParser to add converters (as in https://github.com/OddBloke/jenkins-job-linter/blob/master/jenkins_job_linter/config.py#L26-L39). Those converters via SectionProxy will always type-check as returning Any; this isn't any worse than today.

**Comment by OddBloke** [comment]
Cool, looking now. The test case for this issue (thanks to @reddraggone9 in https://github.com/python/typeshed/pull/1527#issuecomment-321707648):

```py
from configparser import ConfigParser

config = ConfigParser()
config.read_dict({'section': {'key': 'false'}})
assert config['section'].getboolean('key') is False

reveal_type(config['section'].getboolean('key'))
reveal_type(config['section'].getfloat('key'))
reveal_type(config['section'].getint('key'))
```

That currently outputs:

```
1543.py:7: error: Revealed type is 'Any'
1543.py:8: error: Revealed type is 'Any'
1543.py:9: error: Revealed type is 'Any'
```

but should output

```
1543.py:7: error: Revealed type is 'builtins.bool'
1543.py:8: error: Revealed type is 'builtins.float'
1543.py:9: error: Revealed type is 'builtins.int'
```

**Comment by OddBloke** [comment]
The commit above fixes this, but I committed it on top of #1527 so I'll wait for that to be merged to rebase and propose an MP.

**Comment by ambv** [comment]
#1527 is merged.
